packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -736,10 +741,17 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
 	/**
 	 * Submits a new signal to the server
 	 *
-	 * @param message - signal to submit
+	 * @param content - Content of the signal.
+	 * @param targetClientId - When specified, the signal is only sent to the provided client id.
 	 */
-	public submitSignal(message: IDocumentMessage): void {
-		this.emitMessages("submitSignal", [[message]]);
+	public submitSignal(content: IDocumentMessage, targetClientId?: string): void {
+		const signal: ISentSignalMessage = {
+			content,
+			targetClientId,
+		};
+
+		// back-compat: the typing for this method and emitMessages is incorrect, will be fixed in a future PR
+		this.emitMessages("submitSignal", [signal] as any);
```

Comment:
This change assumes PUSH can handle (auto-detect) old and new format, right?
It used to be IDocumentMessage[][], now it's {content: IDocumentMessage, targetClientId?: string}[]

I'm not sure why it was same shape as ops (essentially indicating support for some kind of batching).

Reply:
`IDocumentMessage[][]` has never been the correct typing. `content: IDocumentMessage` is also wrong.
It should be `content: unknown` but I'm not going to introduce a breaking typing change in this PR to resolve this

Reply:
@WillieHabi is trying to correct the types. Are you saying `content: unknown` because no one really cares?
We see that `ISignalEnvelope` is commonly sent. Not sure if that is has always been the case.